src/server.rs

```diff
@@ -599,6 +693,43 @@ impl LanguageServer for LspServer {
         }
         Ok(None)
     }
+    async fn rename(
+        &self,
+        params: lsp::RenameParams,
+    ) -> Result<Option<lsp::WorkspaceEdit>> {
+        let key = params.text_document_position.text_document.uri;
+        let store = self.store.lock().unwrap();
+        if !store.contains_key(&key) {
+            error!(
+                "textDocument/didChange called on unknown file {}",
+                key
+            );
+        }
+        let contents = store.get(&key).unwrap();
```

Comment:
This seems like an unsafe `unwrap()`. Will this potentially panic?

Reply:
In normal circumstances, yes there's the potential for this to panic, but in the code block above, we explicitly check to see if the map has an entry for `key`.

Reply:
Can we refactor this to do both at the same time then? Something like:

```
let contents = match store.get(&key) {
    Some(v) => v,
    None => error!("textDocument/didChange called on unknown file {}", key),
};
```